src/layers/legacy/medVtkInria/vtkImageView/vtkImageView3D.cxx

```diff
@@ -1287,3 +1289,22 @@ medVtkImageInfo* vtkImageView3D::GetMedVtkImageInfo(int layer /*= 0*/) const
 
     return imageInfo;
 }
+
+
+void  vtkImageView3D::initImage(int pi_iLayer)
+{
+    if (pi_iLayer == 0)
+    {
+        double *range = m_poInternalImageFromInput->GetScalarRange();
```

Comment:
is it possible to move these 2 lines in SetFirstLayer? It would be clear

Reply:
It's really make sens to stay in the initialization of TransferFunctions phase.